src/databricks/labs/ucx/source_code/queries.py

```diff
@@ -43,41 +43,56 @@ def schema(self):
         return self._session_state.schema
 
     def lint(self, code: str) -> Iterable[Advice]:
-        for statement in sqlglot.parse(code, read='databricks'):
-            if not statement:
-                continue
-            for table in statement.find_all(Table):
-                if isinstance(statement, Use):
-                    # Sqlglot captures the database name in the Use statement as a Table, with
-                    # the schema  as the table name.
-                    self._session_state.schema = table.name
-                    continue
-                if isinstance(statement, Create) and statement.kind == "SCHEMA":
-                    # Sqlglot captures the schema name in the Create statement as a Table, with
-                    # the schema  as the db name.
-                    self._session_state.schema = table.db
+        try:
+            statements = parse_sql(code, read='databricks')
+            for statement in statements:
+                if not statement:
                     continue
+                yield from self._lint_statement(statement)
+        except SqlParseError:
+            yield Failure(
+                code='table-migrate',
+                message=f"SQL query {code} is not supported yet",
```

Comment:
The code being (not) parsed could be quite long: tens or even hundreds of lines. At the very least for this I'd prefer to see it at the end of the message rather than in the middle.
```suggestion
                message=f"SQL query is not supported yet: {code}",
```